### PR TITLE
[KOGITO-9077] Restoring build-quarkus-app.sh and creating maven project to execute it

### PR DESCRIPTION
**Pull request by davidesalerno**
Related PRs:
- https://github.com/kiegroup/kogito-images/pull/1573
- https://github.com/kiegroup/kogito-images/pull/1578

Many thanks for submitting your Pull Request :heart:! 

Please make sure your PR meets the following requirements:

- [x] You have read the [contributors guide](README.md#contributing-to-kogito-images-repository)
- [x] Pull Request title is properly formatted: `[KOGITO|RHPAM-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a testcase that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster
- [x] You've added a [RELEASE_NOTES.md](RELEASE_NOTES.md) entry regarding this change

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- <b>(Re)run Jenkins tests</b>  
  Please add comment: <b>Jenkins [test|retest] this</b>

- <b>Prod tests</b>  
  Please add comment: <b>Jenkins (re)run [prod|Prod|PROD]</b>
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

**Comment by kie-ci1** [comment]
**PR job** `#811` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/811/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[INFO] Cleaning up running containers and images. Any error here can be ignored<br/>[Pipeline] sh<br/>+ docker ps -a -q<br/>+ tr '\n' ,<br/>[Pipeline] sh<br/>+ docker images -q<br/>+ tr '\n' ,<br/>[Pipeline] sh<br/>+ docker buildx rm mybuilder<br/>ERROR: no builder "mybuilder" found<br/>+ true<br/>[Pipeline] sh<br/>+ docker rm -f binfmt<br/>Error response from daemon: No such container: binfmt<br/>[Pipeline] sh<br/>+ docker context ls<br/>NAME        DESCRIPTION                               DOCKER ENDPOINT               ERROR<br/>default *   Current DOCKER_HOST based configuration   unix:///var/run/docker.sock   <br/>[Pipeline] sh<br/>+ docker buildx inspect<br/>Name:   default<br/>Driver: docker<br/><br/>Nodes:<br/>Name:      default<br/>Endpoint:  default<br/>Status:    running<br/>Buildkit:  v0.11.7-0.20230525183624-798ad6b0ce9f<br/>Platforms: linux/amd64, linux/amd64/v2, linux/amd64/v3, linux/386, linux/arm64, linux/riscv64, linux/ppc64le, linux/s390x, linux/mips64le, linux/mips64, linux/arm/v7, linux/arm/v6<br/>[Pipeline] sh<br/>+ docker buildx ls<br/>NAME/NODE DRIVER/ENDPOINT STATUS  BUILDKIT                              PLATFORMS<br/>default * docker                                                        <br/>  default default         running v0.11.7-0.20230525183624-798ad6b0ce9f linux/amd64, linux/amd64/v2, linux/amd64/v3, linux/386, linux/arm64, linux/riscv64, linux/ppc64le, linux/s390x, linux/mips64le, linux/mips64, linux/arm/v7, linux/arm/v6<br/>[Pipeline] sh<br/>+ docker rm -f registry-5000<br/>Error response from daemon: No such container: registry-5000<br/>[Pipeline] sh<br/>+ docker ps<br/>CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/.cekit/cache<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/811/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

File: maven/build-quarkus-app.sh [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+if [ -z ${quarkus_platform_version} ]; then
+    echo "Please provide the quarkus version"
+    exit 1
+fi
+
+case ${image_name} in
+    "kogito-swf-builder")        ;;
+    "kogito-swf-devmode")
+        quarkus_extensions="${quarkus_extensions},${kogito_swf_devmode_extensions}"
+        quarkus_extensions_2_16="kogito-addons-quarkus-jobs-service-embedded,kogito-addons-quarkus-data-index-inmemory"
```
Comment (radtriste) [inline maven/build-quarkus-app.sh]: should be on line 21

Comment (radtriste) [reply maven/build-quarkus-app.sh]: Please sync with https://github.com/kiegroup/kogito-images/blob/159c6700feebaad2a6014dab59aed58b7611ea5a/modules/kogito-swf/devmode/build-config/module.yaml#LL10C16-L10C57

**Review by radtriste — COMMENTED** [review]
I would put those files into `scripts/logic`. wdyt ?

**Comment by kie-ci1** [comment]
**PR job** `#812` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/812/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[INFO] Cleaning up running containers and images. Any error here can be ignored<br/>[Pipeline] sh<br/>+ docker ps -a -q<br/>+ tr '\n' ,<br/>[Pipeline] sh<br/>+ docker images -q<br/>+ tr '\n' ,<br/>[Pipeline] sh<br/>+ docker buildx rm mybuilder<br/>ERROR: no builder "mybuilder" found<br/>+ true<br/>[Pipeline] sh<br/>+ docker rm -f binfmt<br/>Error response from daemon: No such container: binfmt<br/>[Pipeline] sh<br/>+ docker context ls<br/>NAME        DESCRIPTION                               DOCKER ENDPOINT               ERROR<br/>default *   Current DOCKER_HOST based configuration   unix:///var/run/docker.sock   <br/>[Pipeline] sh<br/>+ docker buildx inspect<br/>Name:   default<br/>Driver: docker<br/><br/>Nodes:<br/>Name:      default<br/>Endpoint:  default<br/>Status:    running<br/>Buildkit:  v0.11.7-0.20230525183624-798ad6b0ce9f<br/>Platforms: linux/amd64, linux/amd64/v2, linux/amd64/v3, linux/386<br/>[Pipeline] sh<br/>+ docker buildx ls<br/>NAME/NODE DRIVER/ENDPOINT STATUS  BUILDKIT                              PLATFORMS<br/>default * docker                                                        <br/>  default default         running v0.11.7-0.20230525183624-798ad6b0ce9f linux/amd64, linux/amd64/v2, linux/amd64/v3, linux/386<br/>[Pipeline] sh<br/>+ docker rm -f registry-5000<br/>Error response from daemon: No such container: registry-5000<br/>[Pipeline] sh<br/>+ docker ps<br/>CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/.cekit/cache<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/812/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Comment by rgdoliveira** [comment]
> I would put those files into `scripts/logic`. wdyt ?

Thanks @radtriste , I've applied your suggestions.

**Comment by kie-ci1** [comment]
**PR job** `#813` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/813/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[INFO] Cleaning up running containers and images. Any error here can be ignored<br/>[Pipeline] sh<br/>+ docker ps -a -q<br/>+ tr '\n' ,<br/>[Pipeline] sh<br/>+ docker images -q<br/>+ tr '\n' ,<br/>[Pipeline] sh<br/>+ docker buildx rm mybuilder<br/>ERROR: no builder "mybuilder" found<br/>+ true<br/>[Pipeline] sh<br/>+ docker rm -f binfmt<br/>Error response from daemon: No such container: binfmt<br/>[Pipeline] sh<br/>+ docker context ls<br/>NAME        DESCRIPTION                               DOCKER ENDPOINT               ERROR<br/>default *   Current DOCKER_HOST based configuration   unix:///var/run/docker.sock   <br/>[Pipeline] sh<br/>+ docker buildx inspect<br/>Name:   default<br/>Driver: docker<br/><br/>Nodes:<br/>Name:      default<br/>Endpoint:  default<br/>Status:    running<br/>Buildkit:  v0.11.7-0.20230525183624-798ad6b0ce9f<br/>Platforms: linux/amd64, linux/amd64/v2, linux/amd64/v3, linux/386<br/>[Pipeline] sh<br/>+ docker buildx ls<br/>NAME/NODE DRIVER/ENDPOINT STATUS  BUILDKIT                              PLATFORMS<br/>default * docker                                                        <br/>  default default         running v0.11.7-0.20230525183624-798ad6b0ce9f linux/amd64, linux/amd64/v2, linux/amd64/v3, linux/386<br/>[Pipeline] sh<br/>+ docker rm -f registry-5000<br/>Error response from daemon: No such container: registry-5000<br/>[Pipeline] sh<br/>+ docker ps<br/>CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/.cekit/cache<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/813/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

**Comment by kie-ci1** [comment]
**PR job** `#814` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/814/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[INFO] Cleaning up running containers and images. Any error here can be ignored<br/>[Pipeline] sh<br/>+ docker ps -a -q<br/>+ tr '\n' ,<br/>[Pipeline] sh<br/>+ docker images -q<br/>+ tr '\n' ,<br/>[Pipeline] sh<br/>+ docker buildx rm mybuilder<br/>ERROR: no builder "mybuilder" found<br/>+ true<br/>[Pipeline] sh<br/>+ docker rm -f binfmt<br/>Error response from daemon: No such container: binfmt<br/>[Pipeline] sh<br/>+ docker context ls<br/>NAME        DESCRIPTION                               DOCKER ENDPOINT               ERROR<br/>default *   Current DOCKER_HOST based configuration   unix:///var/run/docker.sock   <br/>[Pipeline] sh<br/>+ docker buildx inspect<br/>Name:   default<br/>Driver: docker<br/><br/>Nodes:<br/>Name:      default<br/>Endpoint:  default<br/>Status:    running<br/>Buildkit:  v0.11.7-0.20230525183624-798ad6b0ce9f<br/>Platforms: linux/amd64, linux/amd64/v2, linux/amd64/v3, linux/386<br/>[Pipeline] sh<br/>+ docker buildx ls<br/>NAME/NODE DRIVER/ENDPOINT STATUS  BUILDKIT                              PLATFORMS<br/>default * docker                                                        <br/>  default default         running v0.11.7-0.20230525183624-798ad6b0ce9f linux/amd64, linux/amd64/v2, linux/amd64/v3, linux/386<br/>[Pipeline] sh<br/>+ docker rm -f registry-5000<br/>Error response from daemon: No such container: registry-5000<br/>[Pipeline] sh<br/>+ docker ps<br/>CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/.cekit/cache<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/814/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

File: scripts/logic/build-quarkus-app.sh [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+quarkus_platform_groupid="${2}"
+quarkus_platform_version="${3}"
+kogito_version="${KOGITO_VERSION:-${4}}"
+
+# common extensions used by the kogito-swf-builder and kogito-swf-devmode
+quarkus_extensions='quarkus-kubernetes,kogito-quarkus-serverless-workflow,kogito-addons-quarkus-knative-eventing,smallrye-health'
+# extensions that are not available in RHBQ 2.13.x
+quarkus_extensions_2_16="org.kie.kogito:kogito-addons-quarkus-jobs-service-embedded:${kogito_version},org.kie.kogito:kogito-addons-quarkus-data-index-inmemory:${kogito_version}"
+# dev mode purpose extensions used only by the kogito-swf-devmode
+kogito_swf_devmode_extensions='kogito-quarkus-serverless-workflow-devui,kogito-addons-quarkus-source-files'
```
Comment (radtriste) [inline scripts/logic/build-quarkus-app.sh]:
```suggestion
quarkus_extensions='quarkus-kubernetes,kogito-quarkus-serverless-workflow,kogito-addons-quarkus-knative-eventing,smallrye-health'
# dev mode purpose extensions used only by the kogito-swf-devmode
kogito_swf_devmode_extensions='kogito-quarkus-serverless-workflow-devui,kogito-addons-quarkus-source-files'
# devmode extensions that are not available in RHBQ 2.13.x
kogito_swf_devmode_extensions_2_16="org.kie.kogito:kogito-addons-quarkus-jobs-service-embedded:${kogito_version},org.kie.kogito:kogito-addons-quarkus-data-index-inmemory:${kogito_version},org.kie.kogito:kogito-addons-quarkus-fabric8-kubernetes-service-catalog:${kogito_version}"
```

File: scripts/logic/build-quarkus-app.sh [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+if [ -z ${quarkus_platform_version} ]; then
+    echo "Please provide the quarkus version"
+    exit 1
+fi
+
+case ${image_name} in
+    "kogito-swf-builder")        ;;
+    "kogito-swf-devmode")
+        quarkus_extensions="${quarkus_extensions},${kogito_swf_devmode_extensions},${quarkus_extensions_2_16}"
```
Comment (radtriste) [inline scripts/logic/build-quarkus-app.sh]:
```suggestion
        quarkus_extensions="${quarkus_extensions},${kogito_swf_devmode_extensions},${kogito_swf_devmode_extensions_2_16}"
```

**Review by radtriste — CHANGES_REQUESTED** [review]

**Comment by kie-ci1** [comment]
**PR job** `#815` was: **FAILURE**
Possible explanation: Pipeline failure or project build failure

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/815/display/redirect
See console log:
<details>
<summary><b>Console Logs</b></summary>
[INFO] Cleaning up running containers and images. Any error here can be ignored<br/>[Pipeline] sh<br/>+ docker ps -a -q<br/>+ tr '\n' ,<br/>[Pipeline] sh<br/>+ docker images -q<br/>+ tr '\n' ,<br/>[Pipeline] sh<br/>+ docker buildx rm mybuilder<br/>ERROR: no builder "mybuilder" found<br/>+ true<br/>[Pipeline] sh<br/>+ docker rm -f binfmt<br/>Error response from daemon: No such container: binfmt<br/>[Pipeline] sh<br/>+ docker context ls<br/>NAME        DESCRIPTION                               DOCKER ENDPOINT               ERROR<br/>default *   Current DOCKER_HOST based configuration   unix:///var/run/docker.sock   <br/>[Pipeline] sh<br/>+ docker buildx inspect<br/>Name:   default<br/>Driver: docker<br/><br/>Nodes:<br/>Name:      default<br/>Endpoint:  default<br/>Status:    running<br/>Buildkit:  v0.11.7-0.20230525183624-798ad6b0ce9f<br/>Platforms: linux/amd64, linux/amd64/v2, linux/amd64/v3, linux/386<br/>[Pipeline] sh<br/>+ docker buildx ls<br/>NAME/NODE DRIVER/ENDPOINT STATUS  BUILDKIT                              PLATFORMS<br/>default * docker                                                        <br/>  default default         running v0.11.7-0.20230525183624-798ad6b0ce9f linux/amd64, linux/amd64/v2, linux/amd64/v3, linux/386<br/>[Pipeline] sh<br/>+ docker rm -f registry-5000<br/>Error response from daemon: No such container: registry-5000<br/>[Pipeline] sh<br/>+ docker ps<br/>CONTAINER ID   IMAGE     COMMAND   CREATED   STATUS    PORTS     NAMES<br/>[Pipeline] sh<br/>+ rm -rf /home/jenkins/.cekit/cache<br/>[Pipeline] }<br/>[Pipeline] // script<br/>[Pipeline] script<br/>[Pipeline] {<br/>[Pipeline] sh<br/>+ wget --no-check-certificate -qO - https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/kogito-images/815/api/json<br/>[Pipeline] readJSON<br/>[Pipeline] sh<br/>
</details>

File: scripts/logic/build-quarkus-app.sh [hunk 4]
```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env bash
+# Parameters:
+#   1 - image name - can't  be empty.
+#   2 - git target branch - defaults to main
+#   3 - git target uri - defaults to https://github.com/kiegroup/kogito-apps.git
```
Comment (radtriste) [inline scripts/logic/build-quarkus-app.sh]: To be updated

**Review by radtriste — APPROVED** [review]
one documentation comment

File: scripts/logic/build-quarkus-app.sh (hunk 4 above)
Comment (rgdoliveira) [reply scripts/logic/build-quarkus-app.sh]: updated

**Review by radtriste — APPROVED** [review]

**Pull request merged**